### PR TITLE
Expose offset retention time for Consumer API

### PR DESCRIPTION
**Pull request by tjwp**
Any feedback on this change before I submit it to the upstream repo?

Prime: @jturkel 
CC: @will89

### CODE REVIEW

File: lib/kafka/client.rb [hunk 1]
```diff
@@ -216,8 +216,10 @@ def async_producer(delivery_interval: 0, delivery_threshold: 0, max_queue_size:
     #   not triggered by message processing.
     # @param heartbeat_interval [Integer] the interval between heartbeats; must be less
     #   than the session window.
+    # @param offset_retention_time [Integer] the time period that committed
+    #   offsets will be retained, in seconds. If -1, then the broker default is used.
```
Comment (jturkel) [inline lib/kafka/client.rb]: Is there any precedent for using -1 to represent the broker default in ruby-kafka?

**Review by jturkel — APPROVED** [review]
Just one question on the interface

File: lib/kafka/client.rb (hunk 1 above)
Comment (tjwp) [reply lib/kafka/client.rb]: Yes, this is the value that the broker expects, and `ruby-kafka` uses this in the protocol class: https://github.com/zendesk/ruby-kafka/blob/master/lib/kafka/protocol/offset_commit_request.rb#L5

Comment (jturkel) [reply lib/kafka/client.rb]: I was asking about the API exposed by ruby-kafka not the broker. Just curious if they use nil or -1 for scenarios like this.

Comment (tjwp) [reply lib/kafka/client.rb]: No, skimming the code I don't see -1 used to represent the default in any other obvious public APIs. I'll change it to default to nil here.

**Comment by tjwp** [comment]
I'm going to merge this into a `salsify-master` branch and then submit back to the upstream repo.

**Pull request closed without merging**